### PR TITLE
Added initial display labs

### PR DESCRIPTION
**Pull request by agolubev**
Not sure if this is ready to be merged as is so waiting for feedback.
Added three exercises with display: cluster only, sharding and CRDT.
Added as 101-103 exercises.
Put together a simple description with wire connection schemas. Perhaps need to add step by step instructions
Display control software is on Java using Pi4J. It uses a simple image to font logic

### CODE REVIEW

**Comment by agolubev** [comment]
checked on Pi installation with 
- sbt>universal:packageBin
- copy 10X
- run 10x

**Comment by eloots** [comment]
@agolubev Totally missed your review request. Will look into this asap.

**Comment by agolubev** [comment]
Thank you. Let me check. Maybe I forgot to commit something

**Comment by eloots** [comment]
> Thank you. Let me check. Maybe I forgot to commit something
> […](#)

Got exercise 101 running. Updated comment above.

File: project/CommonSettingsDisplay.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+
+import sbt.Keys._
+import sbt._
+
+object CommonSettingsDisplay {
+
+   lazy val commonSettings = Seq (
```
Comment (eloots) [inline project/CommonSettingsDisplay.scala]: This object and its member `commonSettings` seem to be defined but not used?

File: build.sbt [hunk 2]
```diff
@@ -100,3 +105,18 @@ lazy val exercise_050_cluster_cluster_singleton_akka_bootstrap_discovery_via_con
 lazy val exercise_051_cluster_singleton_akka_bootstrap_discovery_via_akka_dns = project
   .configure(CommonSettings.configure)
   .dependsOn(common % "test->test;compile->compile")
+
+lazy val common_display = project.settings(CommonSettingsDisplay.commonSettings: _*)
```
Comment (eloots) [inline build.sbt]: I see that you put a lot of `.jar` files under the `common_display` project. Why not use regular dependencies (declared under `project/Dependencies.scala`)?

They seems to exist on Maven:
```
"com.pi4j" % "pi4j-core" % "1.2"
"com.pi4j" % "pi4j-device" % "1.2"
"com.pi4j" % "pi4j-gpio-extension" % "1.2"
"com.pi4j" % "pi4j-example" % "1.2"
```

Comment (agolubev) [reply build.sbt]: Yeah, when I added them initially several months ago there was pom files but not jars. Now - I checked that jars are available on maven central - will remove jar from repo

File: project/CommonSettingsDisplay.scala (hunk 1 above)
Comment (agolubev) [reply project/CommonSettingsDisplay.scala]: It is used for common_display project. Right the line you put next comment

Idea is to have config common for all display subprojects.

**Comment by agolubev** [comment]
@eloots  updated this branch. Please check when you have time.

File: akka-pi-os.yml [hunk 3]
```diff
@@ -75,6 +75,17 @@ write_files:
     permissions: '0755'
     path: /usr/local/bin/install-java
 
+  - content: |
```
Comment (eloots) [inline akka-pi-os.yml]: @agolubev It might make sense to create a separate `cloud-init` yml file for this set-up. Like `akka-pi-display-os.yml`...
WDYT?

**Comment by agolubev** [comment]
@eloots I was not aware of course management restrictions. Thanks for the explanation

**Comment by agolubev** [comment]
@eloots  add changes per your feedback regarding eliminating commonDisplay

File: akka-pi-os.yml (hunk 3 above)
Comment (agolubev) [reply akka-pi-os.yml]: Well, it's very easy to transform led cluster to oled one if you have panel assembled. I can definitely create `pi-display` one, but as soon as we have one set of exercises and pretty much compatible I would stay with single init.yml

**Comment by eloots** [comment]
@agolubev I can't verify the display based exercises as I don't have access to the OLED display you're using. Looks good though, but wouldn't it be a good idea to migrate the status tracker to Akka Typed?
The base exercise (`base_cluster`) has been migrated and merged on master (exercise 60 at the time of writing). It has a generic `ClusterStatusTracker` that allows for a specific visualiser to subscribe to cluster status updates. I wrote a specific visualiser for the LED strips - you could do the same for the OLED visualiser.
There's also PR #136 which has versions of all but two exercises moved to Akka Typed. If you can do this, we can phase out Akka Classic on this repo. It's ready for review.

**Comment by eloots** [comment]
@agolubev I propose to merge this when you resolve the merge conflicts on `build.sbt`. I would move the version of `akka-pi-os.yml` for the OLED display to `akka-pi-os-display.yml`

**Comment by agolubev** [comment]
@eloots I rebased and resolved all conflicts. Also created akka-pi-os-display.yml. Can you approve it so I can go forward with merge?

**Pull request merged**